plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java

```diff
@@ -191,16 +201,47 @@ public void testCreateTableIfNotExists()
         }
     }
 
-    @Test
-    public void testEmptyProjection()
+    @Test(dataProvider = "supportedTableTypesDDL")
+    public void testEmptyProjection(TableDefinition.Type tableType, String createSql, String dropSql)
     {
-        // Regression test for https://github.com/trinodb/trino/issues/14981
-        try (TestTable table = new TestTable(
-                getQueryRunner()::execute,
-                "test_emtpy_projection",
-                " AS SELECT * FROM region")) {
-            assertQuery("SELECT count(*) FROM " + table.getName(), "VALUES 5");
+        // Regression test for https://github.com/trinodb/trino/issues/14981, https://github.com/trinodb/trino/issues/5635 and https://github.com/trinodb/trino/issues/6696
+        String name = TEST_SCHEMA + ".test_empty_projection_" + tableType.name().toLowerCase(ENGLISH) + randomNameSuffix();
+        try {
+            onBigQuery(createSql);
```

Comment:
nit: I would move `onBigQuery(createSql);` to before `try` block and remove `IF EXISTS` option in drop table statements. It ensures that drop table statement is valid in the succeeded case.